shared/js/utils.js

```diff
@@ -368,19 +368,24 @@ var inChrome = typeof Components != "undefined" && "utils" in Components;
    * Formats a url for display purposes. This includes converting the
    * domain to punycode, and then decoding the url.
    *
-   * @param {String} url The url to format.
-   * @return {Object}    An object containing the hostname and full location.
+   * @param {String}  url     The url to format.
+   * @param {String}  noError Called with a boolean which is true to squash the console error.
+   * @return {Object}         An object containing the hostname and full location.
    */
-  function formatURL(url) {
+  function formatURL(url, noError) {
     // We're using new URL to pass this through the browser's ACE/punycode
     // processing system. If the browser considers a url to need to be
     // punycode encoded for it to be displayed, then new URL will do that for
     // us. This saves us needing our own punycode library.
+    // URL is canonicalizing hostname-only URLs, as GET always has to include a path
+    // and correctly adds a backslash.
```

Comment:
Maybe "Note that URL does canonicalize hostname-only URLs, adding a slash to them, but this is ok for at least HTTP(S) because GET always has to specify a path, which will (by default) be /"

@adamroach Is it reasonable to assume that this is ok for non-HTTP(S) URLs as well?